src/Microsoft.DotNet.Darc/src/DarcLib/Helpers/LocalHelpers.cs

```diff
@@ -138,5 +205,18 @@ public static string ExecuteCommand(string fileName, string arguments, ILogger l
 
             return output;
         }
+
+        private static void ExecuteGitShallowSparseCommand(string fileName, string arguments, ILogger logger, string workingDirectory)
```

Comment:
This name is probably not appropriate anymore:

* it takes an arbitrary command, not just git commands
* It could take commands not related to Shallow clone or sparse checkouts.

It looks like this is just a wrapper for ExecuteCommand that logs the command that will run and throws a DarcException instead of just logging.

Perhaps we can get rid of this function and just add the logging to ExecuteCommand?

Reply:
The idea crossed my mind but didn't go with it because `ExecuteCommand` is used for many other things that we might not want to track (Begin/End scope) for example git authorize which opens a file. Also, in some cases we expect that the returned result is `string.Empty`. Adding logic to deal with these special cases would be worst than just adding a new function I think

Reply:
Can't we just add the begin/end to the caller then?

Reply:
Synced up offline, The name and function can stay as with the latest iteration these will all be git commands, so it makes more sense to keep it.